src/sidebar/services/import-annotations.ts

```diff
@@ -0,0 +1,165 @@
+import type { Annotation, APIAnnotationData } from '../../types/api';
+import { quote } from '../helpers/annotation-metadata';
+import type { SidebarStore } from '../store';
+import type { AnnotationsService } from './annotations';
+import type { ToastMessengerService } from './toast-messenger';
+
+/**
+ * The subset of annotation fields which are preserved during an import.
+ */
+type ImportData = Pick<
+  APIAnnotationData,
+  'document' | 'tags' | 'text' | 'target' | 'uri'
+>;
```

Comment:
This discards fields like `created`, `updated`, `id` etc. which are going to be overwritten by the server. I have gone for an opt-in rather than opt-out approach to keeping fields in this PR, as the majority of top-level fields in exported data are server-set fields which should be discarded.